util/puppet_agent_mgr/mgr_windows.rb

```diff
@@ -4,17 +4,29 @@ class PuppetAgentMgr
       class MgrWindows < MgrV3
 
         # is the agent daemon currently running?
-        # this will require win32/service
         def daemon_present?
-          require 'win32/service'
-          case Win32::Service.status(@puppet_service).current_state
-          when "running", "continue pending", "start pending"
-            true
+          if defined?(Puppet::Util::Windows::Service)
+            # Puppet 7 has built-in support for Windows sevices
+            case Puppet::Util::Windows::Service.service_state(@puppet_service)
+            when :SERVICE_RUNNING, :SERVICE_CONTINUE_PENDING, :SERVICE_START_PENDING
+              true
+            else
+              false
+            end
```

Comment:
change the entire case statement to:

```ruby
[:SERVICE_RUNNING, :SERVICE_CONTINUE_PENDING, :SERVICE_STATE_PENDING].include?(Puppet::Util::Windows::Service.service_state(@puppet_service))
```

Can do the same with the other one just below

Reply:
Done!